app/javascript/controllers/main.js

```diff
@@ -0,0 +1,175 @@
+                    updateCurrency('EUR', $scope.wallet.EUR);
+                    updateCurrency('PLN', $scope.wallet.PLN);
+                };
+                $scope.applyBuyEur = function (currencyBuyEur)
+                {
+                    currencyBuyEur = $scope.getEur.rates[0].bid;
+                    $scope.wallet.EUR -= $scope.moneyEur.value;
+                    $scope.wallet.PLN += $scope.moneyEur.value * currencyBuyEur;
+                    updateCurrency('EUR', $scope.wallet.EUR);
+                    updateCurrency('PLN', $scope.wallet.PLN);
+                };
+
+
+                $scope.moneyGbp = {value: null};
+                $scope.applySellGbp = function (currencySellGbp)
+                {
+                    currencySellGbp = $scope.getGbp.rates[0].ask;
+                    $scope.wallet.GBP += $scope.moneyGbp.value / currencySellGbp;
+                    $scope.wallet.PLN -= $scope.moneyGbp.value;
+                    updateCurrency('GBP', $scope.wallet.GBP);
+                    updateCurrency('PLN', $scope.wallet.PLN);
+
+                };
+                $scope.applyBuyGbp = function (currencyBuyGbp)
+                {
+                    currencyBuyGbp = $scope.getGbp.rates[0].bid;
+                    $scope.wallet.GBP -= $scope.moneyGbp.value;
+                    $scope.wallet.PLN += $scope.moneyGbp.value * currencyBuyGbp;
+                    updateCurrency('GBP', $scope.wallet.GBP);
+                    updateCurrency('PLN', $scope.wallet.PLN);
+                };
+
+
+                /*=======================================
+                 ADD OPACITY TO WALLET
+                 ==================================================*/
+                $scope.myClass = {expose:false};
+                $scope.addClass = function() {
+                    $scope.myClass.expose = true;
+                };
+                $scope.removeClass = function() {
+                    $scope.myClass.expose = false;
+                };
+
+
+
+                /*=======================================
+                 SHOW/HIDE TRANSACTION BOXES SCRIPTS
+                 ==================================================*/
+                $scope.toggleSellUsd = function() {
+                    $scope.sellUsd = !$scope.sellUsd;
+                };
+                $scope.toggleBuyUsd = function() {
+                    $scope.buyUsd = !$scope.buyUsd;
+                };
+
+                $scope.toggleSellEur = function() {
+                    $scope.sellEur = !$scope.sellEur;
+                };
+                $scope.toggleBuyEur = function() {
+                    $scope.buyEur = !$scope.buyEur;
+                };
+
+                $scope.toggleSellGbp = function() {
+                    $scope.sellGbp = !$scope.sellGbp;
+                };
+                $scope.toggleBuyGbp = function() {
+                    $scope.buyGbp = !$scope.buyGbp;
+                };
+            });
+})();
```

Comment:
File looks fine but is very hard to get what is going on, maybe usefull will be separate function with every code execution...